pkg/controller/expcertificates/issuing/issuing_controller.go

```diff
@@ -0,0 +1,396 @@
+	// keystore bundle support.
+	// TODO:
+	experimentalIssuePKCS12 bool
+	// ExperimentalPKCS12KeystorePassword is the password used to encrypt and
+	// decrypt PKCS#12 bundles stored in Secret resources.
+	// This option only has any affect is ExperimentalIssuePKCS12 is true.
+	// TODO:
+	experimentalPKCS12KeystorePassword string
+	// experimentalIssueJKS, if true, will make the certificates controller
+	// create a `keystore.jks` in the Secret resource for each Certificate.
+	// This can only be toggled globally, and the keystore will be encrypted
+	// with the supplied ExperimentalJKSPassword.
+	// This flag is likely to be removed in future in favour of native JKS
+	// keystore bundle support.
+	// TODO;
+	experimentalIssueJKS bool
+	// experimentalJKSPassword is the password used to encrypt and
+	// decrypt JKS files stored in Secret resources.
+	// This option only has any affect is experimentalIssueJKS is true.
+	// TODO;
+	experimentalJKSPassword string
+}
+
+// secretData is a structure wrapping private key, certificate and CA data
+type secretData struct {
+	sk, cert, ca []byte
+}
+
+func NewController(
+	log logr.Logger,
+	client cmclient.Interface,
+	factory informers.SharedInformerFactory,
+	cmFactory cminformers.SharedInformerFactory,
+	recorder record.EventRecorder,
+	clock clock.Clock,
+) (*controller, workqueue.RateLimitingInterface, []cache.InformerSynced) {
+	// create a queue used to queue up items to be processed
+	queue := workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(time.Second*1, time.Second*30), ControllerName)
+
+	// obtain references to all the informers used by this controller
+	certificateInformer := cmFactory.Certmanager().V1alpha2().Certificates()
+	certificateRequestInformer := cmFactory.Certmanager().V1alpha2().CertificateRequests()
+	secretsInformer := factory.Core().V1().Secrets()
+
+	certificateInformer.Informer().AddEventHandler(&controllerpkg.QueuingEventHandler{Queue: queue})
+	certificateRequestInformer.Informer().AddEventHandler(&controllerpkg.BlockingEventHandler{
+		WorkFunc: controllerpkg.HandleOwnedResourceNamespacedFunc(log, queue, certificateGvk, certificates.CertificateGetFunc(certificateInformer.Lister())),
+	})
+
+	// build a list of InformerSynced functions that will be returned by the Register method.
+	// the controller will only begin processing items once all of these informers have synced.
+	mustSync := []cache.InformerSynced{
+		certificateRequestInformer.Informer().HasSynced,
+		secretsInformer.Informer().HasSynced,
+		certificateInformer.Informer().HasSynced,
+	}
+
+	return &controller{
+		certificateLister:        certificateInformer.Lister(),
+		certificateRequestLister: certificateRequestInformer.Lister(),
+		secretLister:             secretsInformer.Lister(),
+		client:                   client,
+		recorder:                 recorder,
+		clock:                    clock,
+
+		//enableSecretOwnerReferences: ctx.CertificateOptions.EnableOwnerRef
```

Comment:
Secret owner references need handling too (in the same way as before).
Also as you noted on the stand up, we need to also support the *existing* method of P12/JKS too. If you could include the "Add JKS and P12 support" bit in a separate commit as well, it will make it easier for us to remove it in future (by looking at just that one commit 😄)